core/src/repair/repair_weighted_traversal.rs

```diff
@@ -281,22 +282,24 @@ pub mod test {
         repairs = vec![];
         outstanding_repairs = HashMap::new();
         slot_meta_cache = HashMap::default();
+        let keypair = Keypair::new();
+        let reed_solomon_cache = ReedSolomonCache::default();
+
         let completed_shreds: Vec<Shred> = [0, 2, 4, 6]
             .iter()
-            .map(|slot| {
-                let parent_offset = u16::from(*slot != 0);
```

Comment:
So my understanding is that we are trying to recreate the case where we are getting the last data shred of a single-shred FEC set in the block of index `slot`.  In merkle shreds, it would mean we make a single FEC set of 32+32, and insert the entire 32-shred series of data shreds to achieve the same result. We simply can not make shred with index 32 and LAST_IN_SLOT, so inserting e.g. index 63 (LAST_IN_SLOT) will prompt blockstore to request repairs for all the other shreds in slot it is now missing, so highest_slot check later will not work.

Reply:
@AshwinSekar  can you confirm that this thinking is correct here?

Reply:
Thank you!